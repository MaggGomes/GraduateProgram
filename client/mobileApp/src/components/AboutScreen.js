import React, { Component } from 'react';
import { StyleSheet, Text, View, StatusBar, TouchableOpacity, Linking } from 'react-native';
import Icons from 'react-native-vector-icons/Ionicons';

/* About Screen */
export default class AboutScreen extends Component {
    static navigationOptions = ({ navigation }) => {
        return {
            title: 'About',
            headerStyle: {
                backgroundColor: '#0145CD'
            },
            headerTintColor: 'white',
            headerLeft: (
                <TouchableOpacity style={{paddingLeft:20}} onPress={() => navigation.navigate('Home')}>
                    <Icons name="md-arrow-back" size={25} color="#ffff" />
                </TouchableOpacity>
            )
        }
    };

    handleClickLinkedin = () => {
        Linking.canOpenURL('https://www.linkedin.com/in/manuelggomes/').then(supported => {
            if (supported) {
                Linking.openURL('https://www.linkedin.com/in/manuelggomes/');
            } else {
                console.log("Don't know how to open URI: " + 'https://www.linkedin.com/in/manuelggomes/');
            }
        });
    };

    handleClickGitHub = () => {
        Linking.canOpenURL('https://github.com/MaggGomes').then(supported => {
            if (supported) {
                Linking.openURL('https://github.com/MaggGomes');
            } else {
                console.log("Don't know how to open URI: " + 'https://github.com/MaggGomes');
            }
        });
    };

    render() {
        return (
            <View style={styles.container}>
                <StatusBar
                    backgroundColor="#01308d"
                    barStyle="light-content"
                />
                <Text style={styles.author}>Manuel Gomes</Text>
                <View style={styles.social}>
                    <TouchableOpacity style={styles.socialButton} onPress={this.handleClickLinkedin}>
                        <Icons name="logo-linkedin" size={60} color="#494949" />
                    </TouchableOpacity>
                    <TouchableOpacity style={styles.socialButton} onPress={this.handleClickGitHub}>
                        <Icons name="logo-github" size={60} color="#494949" />
                    </TouchableOpacity>
                </View>

            </View>
        );
    }
}

const styles = StyleSheet.create({

    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    },
    author: {
      fontSize: 30,
    },
    social: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    },
    socialButton: {
        padding: 20
    }
});
