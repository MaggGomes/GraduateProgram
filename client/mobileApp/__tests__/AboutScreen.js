import React from 'react';
import renderer from 'react-test-renderer';
import AboutScreen from '../src/components/AboutScreen';

describe('AboutScreen component', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<AboutScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});