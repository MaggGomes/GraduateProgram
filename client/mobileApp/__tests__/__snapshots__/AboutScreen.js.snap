// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreen component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    Manuel Gomes
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 20,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#494949",
              "fontSize": 60,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 20,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#494949",
              "fontSize": 60,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
